### PR TITLE
Visitor can see menu

### PR DESCRIPTION
**Pull request by shahin1290**
## Pivotal Tracker link
[Link to PT](https://www.pivotaltracker.com/story/show/162114540)

## Actions taken
Created A feature file -> visitor_can_see_menu.feature
Added feature, user story and scenario
Created Spec file -> dishes.rb
Added index route, controller, action and view
## What we have learned
How to set up feature and run tests
How to create model and controller

### CODE REVIEW

File: features/visitor_can_see_menu.feature [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    Given the following dishes exist
+      | name     | description          | price | category |   
+      | pizza    | italian favorite     | 100   |   main   |
+
+    When I visit the index page
+    Then I should see "pizza"
+    And I should see "italian favorite"
+    And I should see "100"
+    And I should see "main"
+    And show me the page
```
Comment (GergKllai1) [inline features/visitor_can_see_menu.feature]: And show me the page is for debugging purposes, remove it for PR.

Comment (CamronLDNF) [reply features/visitor_can_see_menu.feature]: Done.

File: config/routes.rb [hunk 2]
```diff
@@ -1,3 +1,4 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  get 'dishes/index'
```
Comment (faraznaeem) [inline config/routes.rb]: use resources and specify what action you want to use

File: spec/factories/dishes.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+FactoryBot.define do
+  factory :dish do
+    name "MyString"
+    description "MyText"
+    price 1
+    category "MyString"
```
Comment (faraznaeem) [inline spec/factories/dishes.rb]: You need to use { } brackets for the latest release of factory bot
eg `name {"MyString"}`

File: db/migrate/20181126184505_create_dishes.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+class CreateDishes < ActiveRecord::Migration[5.2]
+  def change
+    create_table :dishes do |t|
+      t.string :name
+      t.text :description
+      t.integer :price
+      t.string :category
```
Comment (faraznaeem) [inline db/migrate/20181126184505_create_dishes.rb:7]: Do you guys have a separate model for the categories how will you play this out when you want to create an association?

File: features/step_definitions/basic_steps.rb [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    visit root_path
+end
+
+Then("I should see {string}") do |value|
+    expect(page).to have_content value
+end
+
+Given("the following dishes exist") do |table|
+    table.hashes.each do |dish|
+        Dish.create!(dish)
```
Comment (faraznaeem) [inline features/step_definitions/basic_steps.rb]: use a factory here

**Review by faraznaeem — CHANGES_REQUESTED** [review]

File: db/migrate/20181126184505_create_dishes.rb (hunk 4 above)
Comment (CamronLDNF) [reply db/migrate/20181126184505_create_dishes.rb:7]: Per our discussion, it's easier to remove this and we'll add it during the association PR instead. Thanks.

File: config/routes.rb (hunk 2 above)
Comment (shahin1290) [reply config/routes.rb]: done

File: spec/factories/dishes.rb (hunk 3 above)
Comment (shahin1290) [reply spec/factories/dishes.rb]: done

File: db/migrate/20181126184505_create_dishes.rb (hunk 4 above)
Comment (shahin1290) [reply db/migrate/20181126184505_create_dishes.rb:7]: It is removed

File: features/step_definitions/basic_steps.rb (hunk 5 above)
Comment (shahin1290) [reply features/step_definitions/basic_steps.rb]: done

File: app/models/dish.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+class Dish < ApplicationRecord
+    validates :name, presence: true
+    validates :description, presence: true
+    validates :price, presence: true
+end
```
Comment (faraznaeem) [inline app/models/dish.rb:3]: `validates_presence_of :name, :description, :price`

**Review by faraznaeem — CHANGES_REQUESTED** [review]

File: app/models/dish.rb (hunk 6 above)
Comment (shahin1290) [reply app/models/dish.rb:3]: done

File: features/step_definitions/basic_steps.rb [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+end
+
+Then("I should see {string}") do |value|
+    expect(page).to have_content value
+end
+
+Given("the following dishes exist") do |table|
+    table.hashes.each do |dish|
+        FactoryBot.create(:dish, dish)
+      end
```
Comment (oliverochman) [inline features/step_definitions/basic_steps.rb]: fix indentation here

File: features/visitor_can_see_menu.feature [hunk 8]
```diff
@@ -0,0 +1,14 @@
+Feature: Visitor can see a menu
+    As a visitor,
+    In order to get an overview of the restaurant products
+    I would like to be able to see a menu
+
+Scenario: View list of dishes on the index page
```
Comment (oliverochman) [inline features/visitor_can_see_menu.feature]: fix indentation for the scenario

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: features/step_definitions/basic_steps.rb (hunk 7 above)
Comment (shahin1290) [reply features/step_definitions/basic_steps.rb]: fixed

File: features/visitor_can_see_menu.feature (hunk 8 above)
Comment (shahin1290) [reply features/visitor_can_see_menu.feature]: fixed

File: features/step_definitions/basic_steps.rb (hunk 7 above)
Comment (shahin1290) [reply features/step_definitions/basic_steps.rb]: fixed

File: features/visitor_can_see_menu.feature (hunk 8 above)
Comment (shahin1290) [reply features/visitor_can_see_menu.feature]: fixed

**Review by oliverochman — APPROVED** [review]

**Pull request merged**